### PR TITLE
Features Miny Chip

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-61

### Description

- Created Miny Chip of History Page

### Screenshots
HistoryInfoCard
HisroryInfocard2

### Tips

```bash
class HomeScreen extends StatefulWidget {
  @override
  State<HomeScreen> createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  bool isSelected = false;
  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Scaffold(
      backgroundColor: theme.colors.light,
      body: Padding(
        padding: const EdgeInsets.symmetric(vertical: 400),
        child: Center(
          child: MinyChip(
            label: 'January',
            selected: isSelected,
            onSelected: (value) {
              setState(() {
                isSelected = value;
              });
            },
          ),
        ),
      ),
    );
  }
}
```

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [x] All items in the author checklist is checked or not applicable
- [x] All files including .lock and assets are reviewed
- [x] Reasonable UI flows/cases related to the change are manually tested
- [x] No commented-out code or unnecessary debug statements.
- [x] No spelling or grammatical errors in code comments or documentation.
- [x] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [x] The branch is up-to-date with the target branch (e.g., staging).
- [x] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

**Review by satyabratanayak — DISMISSED** [review]

File: lib/miny_chip.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+              ),
+            ),
+            _buildSelectedIcon(theme),
+          ],
+        ),
+      ),
+    );
+  }
+
+  RenderObjectWidget _buildSelectedIcon(ThemeData theme) {
```
Comment (quiteachaos) [inline lib/miny_chip.dart]: change it to `Widget` instead of RenderObjectWidget

File: lib/miny_chip.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  ///  changes.
+  const MinyChip({
+    super.key,
+    required this.label,
+    required this.selected,
+    this.onSelected,
+  });
+
+  @override
+  _MinyChipState createState() => _MinyChipState();
```
Comment (quiteachaos) [inline lib/miny_chip.dart]: update the statement to 
`State<MinyChip> createState() => _MinyChipState();`

Comment (krushnakantanayak) [reply lib/miny_chip.dart]: updated .

File: lib/miny_chip.dart (hunk 1 above)
Comment (krushnakantanayak) [reply lib/miny_chip.dart]: Changed

**Review by quiteachaos — APPROVED** [review]

**Review by satyabratanayak — APPROVED** [review]

**Pull request merged**